erpnext/controllers/buying_controller.py

```diff
@@ -704,6 +706,72 @@ def validate_items(self):
 		else:
 			validate_item_type(self, "is_purchase_item", "purchase")
 
+	def update_serial_items_table(self, update = False):
+		# updates the serials if auto created from sle in the document
+		# needs cleanup
+		if self.serial_items and update:
+			for item in self.items:
+				if item.serial_no:
+					for serial in item.serial_no.split('\n'):
+						for serial_item in self.serial_items:
+							if serial_item.item_name == item.item_code and not serial_item.serial_no\
+								and serial_item.type == 'Accepted':
+								serial_item.serial_no = serial.upper()
+								frappe.db.set_value(serial_item.doctype, serial_item.name, 'serial_no', serial.upper())
+								break
+				if item.rejected_serial_no:
+					for serial in item.rejected_serial_no.split('\n'):
+						for serial_item in self.serial_items:
+							if serial_item.item_name == item.item_code and not serial_item.serial_no\
+								and serial_item.type == 'Rejected':
+								serial_item.serial_no = serial.upper()
+								frappe.db.set_value(serial_item.doctype, serial_item.name, 'serial_no', serial.upper())
+								break
+			return
```

Comment:
- This is way too much indentation. Im going to leave it at that. (throughout this PR)
- The lone return indicates that this could maybe be a different function.

Reply:
Yea I'm trying to find a better way to do this.